src/gui/sdlmain.cpp

```diff
@@ -2899,27 +2877,57 @@ static SDL_Point window_bounds_from_label(const std::string& pref,
 	const int percent = [&] {
 		if (pref.starts_with('s')) {
 			return SmallPercent;
+
 		} else if (pref.starts_with('m') || pref == "default" ||
 		           pref.empty()) {
 			return MediumPercent;
+
 		} else if (pref.starts_with('l')) {
 			return LargePercent;
+
 		} else if (pref == "desktop") {
 			return 100;
+
 		} else {
-			LOG_WARNING("DISPLAY: Invalid 'windowresolution' setting: '%s', "
-			            "using 'default'",
-			            pref.c_str());
+			LOG_WARNING(
+			        "DISPLAY: Invalid 'windowresolution' setting: '%s', "
+			        "using 'default'",
+			        pref.c_str());
+
+			set_section_property_value("sdl", "windowresolution", "default");
 			return MediumPercent;
 		}
 	}();
 
+	const auto desktop = get_desktop_size();
+
 	const int w = ceil_sdivide(desktop.w * percent, 100);
 	const int h = ceil_sdivide(desktop.h * percent, 100);
 
 	return {w, h};
 }
 
+static SDL_Point parse_window_resolution_from_conf(const std::string &pref)
+{
+	int w = 0;
+	int h = 0;
+	const bool was_parsed = sscanf(pref.c_str(), "%dx%d", &w, &h) == 2;
```

Comment:
Small gripe but it would be nice if this worked regardless of whether the x is capitalized.  I tested and a capital X does not work here.

Reply:
I agree, it's an easy and useful addition, so doing it.